### PR TITLE
6274: Login validation error message fix

### PR DESCRIPTION
**Pull request by RuthBerry96**

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationSummaryTagHelper.cs [hunk 1]
```diff
@@ -77,14 +76,10 @@ private TagBuilder GetErrorListBuilder()
...
             {
-                var redirectAttribute = ViewContext.ViewData.Model.GetType().GetProperty(model.Key)
-                    ?.GetCustomAttributes<SummaryAnchorAttribute>().FirstOrDefault();
-
-                var redirect = redirectAttribute?.Link ?? model.Key;
-
                 foreach (var error in model.Value.Errors)
                 {
-                    var listItem = GetListItemBuilder(redirect, error.ErrorMessage);
+
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationSummaryTagHelper.cs:81]: Spacing?

File: src/NHSD.BuyingCatalogue.Identity.Api/Styles/login.scss [hunk 2]
```diff
@@ -9,3 +9,7 @@
 .bc-c-login-page-error-box-padding {
     white-space: pre-wrap;
 }
+
+.input-validation-error {
```
Comment (kat1906) [inline src/NHSD.BuyingCatalogue.Identity.Api/Styles/login.scss:13]: have you used this class anywhere?

Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Identity.Api/Styles/login.scss:13]: Yes, it gets used when there is an error on the input. (If you launch the login page, then submit with an error, inspect the input box)

File: src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationSummaryTagHelper.cs (hunk 1 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationSummaryTagHelper.cs:81]: 👍

File: src/NHSD.BuyingCatalogue.Identity.Api/Styles/login.scss (hunk 2 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Identity.Api/Styles/login.scss:13]: After speaking with @kat1906 going to use the nhsuk-input--error class

File: src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationInputTagHelper.cs [hunk 3]
```diff
@@ -141,6 +141,13 @@ private TagBuilder GetInputBuilder()
             }
 
             builder.AddCssClass(TagHelperConstants.NhsInput);
+
+            var modelStateErrors = ViewContext.ViewData.ModelState[For.Name]?.Errors;
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationInputTagHelper.cs:145]: There are now two places the class is trying to find out whether or not a particular field is associated with any errors in the model state. Would it be possible to refactor this so the logic is not duplicated?

Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationInputTagHelper.cs]: 👍

**Pull request merged**